src/m365/tenant/commands/service/service-report-historicalservicestatus.ts

```diff
@@ -0,0 +1,112 @@
+}
+
+interface Options extends GlobalOptions {
+  workload?: string;
+}
+
+class TenantServiceReportHistoricalServicestatusCommand extends Command {
+  public get name(): string {
+    return `${commands.TENANT_SERVICE_REPORT_HISTORICALSERVICESTATUS}`;
+  }
+
+  public get description(): string {
+    return 'Gets the historical service status of the Office 365 Services of the last 7 days from the Office 365 Management API.';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.workload = args.options.workload;
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    if (this.verbose) {
+      cmd.log(`Getting the historical service status of the Office 365 Services of the last 7 days from the Office 365 Management API.`);
+    }
+
+    const serviceUrl: string = 'https://manage.office.com/api/v1.0';
+    const statusEndpoint: string = typeof args.options.workload !== 'undefined' ? `ServiceComms/HistoricalStatus?$filter=Workload eq '${encodeURIComponent(args.options.workload)}'` : 'ServiceComms/HistoricalStatus';
+    const tenantId: string = Utils.getTenantIdFromAccessToken(auth.service.accessTokens[auth.defaultResource].value);
+
+    const requestOptions: any = {
+      url: `${serviceUrl}/${tenantId}/${statusEndpoint}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      json: true
+    };
+
+    request.get(requestOptions)
+      .then((res: any): void => {
+        if (args.options.output === 'json') {
+          cmd.log(res);
+        }
+        else {
+          cmd.log(res.value.map((r: any) => {
+            return {
+              WorkloadDisplayName: r.WorkloadDisplayName,
+              StatusDisplayName: r.StatusDisplayName,
+              StatusTime: r.StatusTime
+            }
+          }));
+        }
+
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-w, --workload [workload]',
+        description: 'Allows retrieval of the historical service status of only one particular service. If not provided, the historical service status of all services will be returned.'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public commandHelp(args: {}, log: (help: string) => void): void {
+    const chalk = vorpal.chalk;
+    log(vorpal.find(this.name).helpInformation());
+    log(
+      `  Remarks:
+
+    To get the name of the particular workload for use with the ${chalk.grey('workload')}
+    option, execute ${chalk.grey(`m365 ${commands.TENANT_STATUS_LIST} --output json`)} and get
```

Comment:
That is a nice suggestion indeed. You don't need the end users to think of another command for getting the workload information. That is nice!
Just being a devil's advocate here, in this particular scenario, don't you think the JSON results would be a bit overwhelming, since we have the results for last 7 days? 
Plus, we had told that we would be using `tenant service list` as per [This Conversation](https://github.com/pnp/cli-microsoft365/pull/1785#discussion_r487701751) . Thoughts on that If I may ask.

Reply:
In the current state, neither approach is perfect. One requires looking up another command, the other plowing through a lot of output. I'm also not sure to what extent we can rely on both operations using the same workload ID. They are a part of the same API but it wouldn't be the first time that not everything is a consistent as you'd want it to be. I'd say, let's stay with the current approach for now, and think how we could improve it in the future. Deal? 🙂

Reply:
Even I had the similar thought about the approach which we can follow. I am with you on your suggestion since neither approach is not perfect or consistent. Later stage if we have updates on the API's we can always come back and enhance our filters to sharpen the reporting from our side too.
Accepting the deal. 👍